### PR TITLE
CSHARP-3049: Update docs for 2.11.0-beta1

### PR DESCRIPTION
**Pull request by vincentkam**
Using https://github.com/vincentkam/mongo-csharp-driver/pull/67 as a guide for the last time we did a beta release :)

### CODE REVIEW

File: Docs/reference/content/what_is_new.md [hunk 1]
```diff
@@ -7,6 +7,35 @@ title = "What's New"
   identifier = "What's New"
   pre = "<i class='fa fa-star'></i>"
 +++
+## What's New in 2.11.0-beta1
+
+Some of the changes in 2.11.0-beta1 include:
```
Comment (vincentkam) [inline Docs/reference/content/what_is_new.md:12]: Taken from the release notes

**Comment by vincentkam** [comment]
Ready for review!

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

**Review by rstam — APPROVED** [review]
LGTM

**Pull request merged**